lib/fog/compute/ecloud.rb

```diff
@@ -266,8 +266,97 @@ def default_organization_uri
         end
       end
 
+      # Custom errors specific to our implementation
+      module Errors
```

Comment:
My only concern here is this module. This class is growing too big and maybe we could extract it to another file and load it using `autoload`. What do you think?